File: peer_lib/java/PeerNode.java
package org.koalaui.arkoala;

import java.util.Map;

public class PeerNode {
    public NativePeerNode peer;
    public PeerNode(ArkUINodeType type, int flags) {
        // TODO: use NativeModule._GetNodeFinalizer()
        this.peer = new NativePeerNode(42, 0);
    }
    void applyAttributes(Map<String, Object> attrs) {}
}

File: peer_lib/ts/PeerNode.ts
import { int32 } from "@koalaui/common"
import { pointer } from "@koalaui/interop"
import { Finalizable } from "./Finalizable"
import { nativeModule } from "./NativeModule";

export class NativePeerNode extends Finalizable {
}

export class PeerNode {
    peer: NativePeerNode
    constructor(type: number, flags: int32) {
        // TODO: type and flags
        this.peer = new NativePeerNode(BigInt(42), getNodeFinalizer())
    }
    applyAttributes(attrs: Object) {}
}


function getNodeFinalizer() : pointer {
    return nativeModule()._GetNodeFinalizer()
}
